packages/babel-types/src/clone/cloneNode.js

```diff
@@ -0,0 +1,50 @@
+import { NODE_FIELDS } from "../definitions";
+
+const has = Function.call.bind(Object.prototype.hasOwnProperty);
+const isNode = obj => obj && has(NODE_FIELDS, obj.type);
+
+function cloneIfNode(obj) {
+  return isNode(obj) ? cloneNode(obj) : obj;
+}
+
+function cloneIfNodeOrArray(obj) {
+  return Array.isArray(obj) ? obj.map(cloneIfNode) : cloneIfNode(obj);
```

Comment:
I don't think we have any types that are arrays of non-Node types. As it is, this will quietly fail to clone

```
{

Reply:
Expanding on this, I wonder if instead we should just make `cloneIfNode` do 

```
return obj && typeof obj === "object" && typeof obj.type === "string" ? cloneNode(obj) : obj;